api/src/list/list.service.ts

```diff
@@ -128,4 +130,28 @@ export class ListService {
       isShared,
     );
   }
+
+  async cloneList(listId: string, userId: string, name?: string) {
+    const { count } = await this.movieService.getMovies({
+      per_page: 0,
+      page_number: 0,
+      listId,
+    });
+    const originalList = await this.getList(listId);
+    const clonedList = await this.createList(name || originalList.name, userId);
+
+    if (count === 0) {
+      return clonedList;
+    }
+
+    const { movies } = await this.movieService.getMovies({
+      per_page: count,
+      page_number: 0,
+      listId,
+    });
+
+    await this.movieService.createMovies(movies, clonedList.id);
```

Comment:
I think we can probably do all this in the controller, and we wouldn't even need a clone method in the service.